### ISSUE TITLE
Prescription Carve-out requirements - BPMS-I-16 AB-172 EA-249 AB-194

### ISSUE THREAD

**Issue by g4gaurang** [willdwyer/bcbsmaissuestracker#256]
We would like to get a level of effort and timeline assessment on the Prescription Carve-out requirements for SBCs. We have shared these requirements with you in the past. You can also find them below: (click link)

https://drive.google.com/file/d/0B_7OzEEzvKXzUEFTb3IxdnZxcGM/view?usp=sharing
https://drive.google.com/file/d/0B_7OzEEzvKXzZ244WDZ2bXNNZXc/view?usp=sharing

[Requirement for Prescription CarveOuts.docx | ](https://files.zenhub.io/56018b92181c0a74779a21ab)

[RxCarveOutExampleTiebackToRequirements.doc | ](https://files.zenhub.io/56018bc1181c0a74779a21b0)

Let us know if you have questions.

@markderg @jddecoste

**Comment by michparrish** [comment]
@g4gaurang the docs are not loading for me. I've tried the hyperlinks above and tried to open in Chrome directly - can you please forward them via email? Thank you

**Comment by g4gaurang** [comment]
> @g4gaurang the docs are not loading for me. I've tried the hyperlinks above and tried to open in Chrome directly - can you please forward them via email? Thank you

The links should work now. Seems like there was a temporary issue accessing the google cloud.

**Comment by michparrish** [comment]
https://drive.google.com/open?id=0BwDpjEumPUB7MkE3aEtoMUpvcUU

let me know if this doesn't work - this is the first time I've used the google drive...

**Comment by g4gaurang** [comment]
> https://drive.google.com/open?id=0BwDpjEumPUB7MkE3aEtoMUpvcUU
> 
> let me know if this doesn't work - this is the first time I've used the google drive...

@michparrish, you will need to provide permissions for access to your documents on google drive. An alternate to this is if you put it on the Highroads system and copy the direct hyperlink to it, that will work as well. Here is the format you can follow - 

![image](https://cloud.githubusercontent.com/assets/10272072/10107329/4e570a50-6388-11e5-9208-ee5cbd485c1e.png)

just replace the text in the parenthesis and leave the parenthesis as is.

So just for example-
[Document Name](https://bcbsma.highroads.com/ie3/control/sso/entrance?parameter=doIMBanner:true&parameter=referer:/homepagebullseye&parameter=initReferenceName:800281HR_Access_Blue_NE_Basic_2000_80-0117LGASC1-1-16&parameter=initReferenceId:3444515&parameter=initIsDocument:true&targetURL=int_cms_browser)

**Comment by michparrish** [comment]
I can't copy the Send Link that I have to the document since you all use SSO (different URL). 

Sorry for the confusion here - where in Google Drive do I update permissions? It's not obvious.

**Comment by g4gaurang** [comment]
You can also use Pclient5 to host the document and share the link - 

Here is how to do it on google drive - 
![image](https://cloud.githubusercontent.com/assets/10272072/10107819/8bff0bd4-638b-11e5-9658-56ab2355da6a.png)

**Comment by michparrish** [comment]
Thank you!

https://drive.google.com/file/d/0BwDpjEumPUB7MkE3aEtoMUpvcUU/view?usp=sharing

this work?

**Comment by g4gaurang** [comment]
> Thank you!
> 
> https://drive.google.com/file/d/0BwDpjEumPUB7MkE3aEtoMUpvcUU/view?usp=sharing
> 
> this work?

Yes, It did work. I was able to access and download it and I noticed this document is not for the prescription drug carve-out requirement for which this request is opened.  It may belong to #251 instead. Also, I see that you have used an enhancement template to capture information - It may work for the HR team but BCBSMA has so far not agreed to contribute to and use it as a basis of agreement and approval. Also, I noticed that on #251, you have noted 72 hours but have quoted 75 hours on the attachment.

@jddecoste @kselvo01

**Comment by michparrish** [comment]
oh dear, my bad. it does belong to 251 - I'll move it. My apologies!

**Comment by g4gaurang** [comment]
@michparrish @markderg  Do you have LOE and timing on this requirement ?

**Comment by michparrish** [comment]
Hi Gaurang,
In the meeting last week, we stated that we would present the options along with risk and LOE to your team by 10/2. Mark has set up a meeting for 2:00 on Friday, 10/2.

**Comment by michparrish** [comment]
Added enhancement request # to name as needed to submit enhancement request to assess LOE for last requirement of new mapping from doc gen to 'SBCs (Word-Carve Out) Final' folder. This will be reviewed in the upcoming sprint which starts on 9/30.

**Comment by michparrish** [comment]
correction to the sprint start date - 10/7

**Comment by michparrish** [comment]
updated requirements document that was reviewed on 10/2:
https://drive.google.com/file/d/0BwDpjEumPUB7eWE5a1dpMEtYYmc/view?usp=sharing

**Comment by michparrish** [comment]
AB-172 will be included in next build, we are pending a deployment schedule to pclient5 for testing - this is for the ability for us to map to a new set of Carve-out-specific templates. I will update the item when this is deployed and tested with success
BPMS-I-16 - pending LOE for the Doc Gen page updates to allow users to select Carve Out denotations so the generated document routes correctly to the SBC-Carve Out folder in plan masters

**Comment by michparrish** [comment]
AB-172 was moved to Prod 11/27. No testing required as this was an internal bug to allow us to properly map templates by more attributes.
Still pending BPMS-16. Meeting scheduled for week of 12/7 with Development to assess request.

**Comment by TimHeasley** [comment]
Total hours for the HighRoads product & services team is 60 hours.

**Comment by g4gaurang** [comment]
We approve the 60 hour estimate. Please provide a definitive sprint release estimate.

**Comment by michparrish** [comment]
Submitted acceptance to Dev - will report back when timeline for work is provided to Services team.

**Comment by TimHeasley** [comment]
Completed in Dev and ready for testing by QA team. We will work with QA testing to try and secure a Production date by end of January.

**Comment by TimHeasley** [comment]
Other 2 enhancements have taken priority to complete this week.  Carve out enhancement will be completed in February.

**Comment by TimHeasley** [comment]
BCBSMA team has approved request to refresh PCLIENT5. HighRoads team plans on having enhancement ready for BCBSMA team to test on March 2nd.

**Comment by willdwyer** [comment]
Hi Tim,

Just assigning a deadline label of March 2nd to this item, so HighRoads is aware this must be ready for test by then.

**Comment by willdwyer** [comment]
Hi,

Just following up on this, as March 2nd is 2 days away.  Is HighRoads close to completing this?

**Comment by TimHeasley** [comment]
Will,
We are currently in the process of updating the templates and workbooks. 
Our goal is still for you to be able to test on March 2nd.
Thanks,
Tim

**Comment by TimHeasley** [comment]
Hi Will,
He have completed our internal testing and the enhancement is now ready for you to test in PClient5.
Please reference the requirements document above for the enhancement updates.
Let me know if you have any questions.
Thanks,
Tim

**Comment by willdwyer** [comment]
Hi Tim,

I have tested this enhancement, per the requirements listed, and half of them have been filled, while the other half have not been fulfilled.  Our requirements in the document were as follows (below).

3, 5, 6, 8 and 9 have been fulfilled.

However, 1, 2, 4 and 7 have not been met, per the testing.
1. The system should remove the BCBSMA Logo from page 1 of the output SBC document 
2. The system should remove copyright/registered wording and licensee language on the both the first and last page footers of the SBC document.
3. The system should remove the QR code from footer of the first page of the SBC document 
4. The system should remove the BCBSMA hyperlink  under the If you need drugs to treat your illness or condition Common Medical Event  and leave it as www.________.com
5. The system should indicate “Not covered” as cost share across all tiers/cost columns across Generic drugs, Preferred, Non-preferred and Specialty drugs services under the If you need drugs to treat your illness or condition Common Medical Event. In addition, the system should also  indicate “---  none ---”  across the Limitations and Exceptions for those services
6. The system should move “Prescription drugs” to the Services Your Plan Does NOT Cover box in the appropriate order.
7. The system should modify the Minimum Value Standard language to indicate “Does not meet”
   "8. The system should pass the values “Not covered” as benefit input for 
   i. Generic Drugs - Mail Order, Preferred Brand Drugs – Retail and Preferred Brand Drugs - Mail Order services on the Type II Diabetes coverage example calculator. 
   ii. Generic Drugs - Retail Cost Sharing on the Maternity Calculator"
8. The system should generate Word versions of the Carve-out SBC and store the output document in the SBCs (Word – Carve Out) Final folder for that account in the content library

I generated a carve-out document, and the SBC still had the: (1) BCBSMA logo on page 1, (2) copyright/registered wording and licensee language on both the first and last page footers; (4) BCBSMA hyperlink in the Rx drug Common Medical Event section and (7) Minimum Value Standard language said "Does Meet" instead of "Does Not Meet" (as it should), along with the Does Meet disclosure instead of the Does Not Meet disclosure.  I have also attached this sample SBC.

Please make these adjustments, and we will retest.
[sbc_preview-15.pdf](https://github.com/willdwyer/bcbsmaissuestracker/files/155693/sbc_preview-15.pdf)

**Comment by TimHeasley** [comment]
Hi Will,
I just went into the plan you used "3539157" in PClient5.
On the Plan Details Tab, Rx Carve Out was No.  Please change this to Yes and you will see the updates for 1, 2, 4 and 7.
Thanks,
Tim

**Comment by willdwyer** [comment]
Tim,

So, I re-generated with checking Yes on the Plan Details Tab, and see that requirements 1, 2 and 4 have all been met.  However, requirement 7 has still not completely been fulfilled (see attached).

On this SBC, the Minimum Value Standard language has been changed to "Does Not Meet", which is correct, but the disclosure does not match the language.  The disclosure at the end is MCC-Does Meet disclosure.  This would have to be changed in the enhancement.
[sbc_preview-21.pdf](https://github.com/willdwyer/bcbsmaissuestracker/files/155831/sbc_preview-21.pdf)

Also, it is not very convenient/practical for the user to have to click 2 options for Carve-Outs (both in the Plan Details and Misc Fields tab.  Can they consolidate this into one?  The user would never click just one, and leave out the other.  It would always be both yes, or both no.

**Comment by TimHeasley** [comment]
Will,
Thank you for testing. 
I will look into these items and let you know the outcome.
Tim

**Comment by willdwyer** [comment]
Hi Tim,

Do you have an estimate as to when requirement 7 will be fulfilled completely, so we can retest?

**Comment by sayotte7701** [comment]
Hi Will,

I'm going to look at this now.   thanks

**Comment by willdwyer** [comment]
Hi Steve, 

Do you have an update on this?  We want to retest this ASAP.

Thanks,

**Comment by TimHeasley** [comment]
Will,
I am currently working on updating the disclosures so that they appear correctly in PClient5. 
As soon as it is ready for retest I will let you know.
Thanks,
Tim

**Comment by willdwyer** [comment]
Hi Tim, do you have an ETA on when the disclosures will be updated?

**Comment by michparrish** [comment]
Hi Will,
One of the requirements for this item, noted at the bottom of the requirements document attached to this item, was that when Carve Out = Yes, the option for Disclosures is greyed out as there should not be Disclosures generated with a Carve out plan. This code does work. Therefore, if you were getting a disclosure, then you were running this on a Carve Out = No plan and when this happens, you are using the original templates, which pull the 'does not meet' from the plan design. So you would have to change the value in the Configurator > Misc Fields tab to 'does not meet' in order to get the right Disclosure.
Please retest to confirm that Disclosures are not available for Carve Out = Yes plans and that things are working as normal for Carve Out = No plans.
Thank you.

**Comment by willdwyer** [comment]
Hi Michelle,

Just to follow up on the call - I just generated a Carve-Out plan, and the areas highlighted in red (BCBSMA logo, registered wording) should be removed for ALL Carve-Out plans (whether it is a Word, PDF, Draft, Final, etc.)

Please make sure this is completely globally with the enhancement

![screen shot 2016-03-18 at 3 42 30 pm](https://cloud.githubusercontent.com/assets/13453008/13890272/f44e1f62-ed20-11e5-9624-0d7b2675ca2a.png)

**Comment by michparrish** [comment]
Per conversations with Will today and Friday, 3/18, a new enhancement request has been opened for Dev, EA-249.

This item was opened due to the original requirements instructing PDF to not be available when generating Final Carve Out documents, only Word. Will has confirmed that Final Carve Out docs can be in both Word and PDF and should be routed as such:
If it is a Carve Out = Yes, Final, PDF, map to: 1-SBCs (PDF) – Final
If it is a Carve Out = Yes, Final, Word, map to: 2-SBCs (Word – Carve Out) – Final
If it is a Carve Out = Yes, Draft, PDF or Word, map to: 2-SBCs (PDF and Word) – Draft (added in just for consistency)

Since the code from the original requirements is completed and tested successfully, we will still push this code into Prod with the other code that is coupled with it, but BCBS will not utilize this functionality until the updated PDF code is in place.
Associated items with code pushed to Prod:
260, 293, 335, 336

@willdwyer please add your confirmation for us to move forward with the code as is. We will continue to post updates to this item as we move the updated code through the process.

Thanks!

**Comment by willdwyer** [comment]
You have my confirmation.  Please proceed.

**Comment by willdwyer** [comment]
Please provide an ETA of when this will be uploaded today.

**Comment by michparrish** [comment]
Hi Will,
The full code push won't be finished today. Now that you've finished your testing and approval for all of the code updates, we need to go back to the process and have IT push the code to a QA env for a final 'sanity check' and then we'll have a go/no go meeting. We'll likely push that night. It's just a scheduled thing that we need to set up now. I'll let you know when we schedule the push to Prod. We're hoping by end of this week but we need to ensure we have the availability of everyone in play. 
Thanks!

**Comment by michparrish** [comment]
just a note that the updated template mapping is updated in pclient5 for your review as well. Understood that we will NOT be pushing this configuration into Prod, but it's there for you to test.

**Comment by willdwyer** [comment]
Michelle,

As I mentioned, we need enhancements #260 #293 and #335 pushed to production ASAP. New hires have already been trained on these, and they must start the process in production soon.  Please treat this as an urgent matter.

**Comment by michparrish** [comment]
email exchange between Michelle and Will - just documenting:

The requirements below look correct.  The Rx Carve Out region on the Generator screen is used to determine if a Word document is mapped as 2-SBCs (Word – Carve Out) - Final or 2-SBCs (Word) - Final.

In an ideal world, we would like the Carve Out option to be isolated to 1 sole button.  We currently have 3 different options to select a Carve Out (Plan Details, Misc Fields and Generator screen), which may cause issues if not all buttons are selected by a user in a carve out plan.

Just a thought – You had mentioned that the Carve Out button on the Plan Details tab was to determine attributes, while the Carve Out button on the Misc Fields tab was to determine logic.  Can there be a relationship/correlation between the Carve Out attribute and designation of subfolder for plan(s) being generated?  For example, if Carve Out Yes on plan details tab is selected, it would function the same as the Rx Carve Out plan button on the generator?  Please let me know.

 –––––––––––––––––––––––––––––––––––––––––––
William Dwyer
Business Analyst
Compliance & Benefit Communications
Blue Cross Blue Shield of Massachusetts
One Enterprise Drive, Quincy, MA 02171-2126

617.246.3827

From: Michelle Parrish mparrish@highroads.com
Date: Thursday, March 24, 2016 at 3:14 PM
To: bcbsma William.Dwyer@bcbsma.com
Cc: Tim Heasley theasley@highroads.com, Steven Ayotte sayotte@highroads.com, Mark Der Garabedian mdergarabedian@highroads.com
Subject: Rx Carve Out Requirements

Hi Will,

I’m working with Dev right now and I wanted to just confirm the requirements again:

Rx Carve Out: Yes/No on the document generation screen was implemented to incur the proper mapping of the document generated when it was a Final Word doc, which would map the document to the “2-SBCs (Word – Carve Out) – Final” folder.

Therefore:
Rx Carve Out: Yes/No field will be disabled when user selects Draft – all draft documents, Carve Out or not, will continue to map to the same folder - 2-SBCs (PDF and Word) – Draft’
Rx Carve Out: Yes/No will be enabled on Final for both Word and PDF
Final Word documents route to ‘2-SBCs (Word – Carve Out) – Final’
Final PDF documents route to ‘1-SBCs (PDF) – Final’
Additionally:
Rx Carve Out: Yes/No field will be disabled when Disclosures = Yes
Disclosures field will be disabled when Rx Carve Out = Yes

Thank you so much for re-confirming. This was stated a little differently than how I presented it, but it meets the exact same end result.

**Comment by TimHeasley** [comment]
Code has been moved to Production, taking into consideration previous information.

"Since the code from the original requirements is completed and tested successfully, we will still push this code into Prod with the other code that is coupled with it, but BCBS will not utilize this functionality until the updated PDF code is in place."

**Comment by TimHeasley** [comment]
Additional enhancement request has been sent to development team and is being worked on.

**Comment by michparrish** [comment]
EA-249 has been released to pclient5 for testing.
This is specifically for the 'Rx Carve Out: Yes/No' radio button on the Doc Gen screen of the Configurator.
Requirement - when Rx Carve Out = Yes, and document is a Final Word document with no Disclosures, the document should map to the 2-SBCs (Word - Carve Out) - Final.
Original requirements rolled back/removed - PDFs are now allowed when creating a Carve Out doc, Disclosures are now allowed when creating a Carve Out doc.

Testing results are as follows:
Draft + No Carveout + No Disclosures + Word = 2-SBCs (PDF and Word) - Draft
Draft + No Carveout + No Disclosures + PDF = 2-SBCs (PDF and Word) - Draft
Final + No Carveout + No Disclosures + Word = 1-SBCs (Word) - Final
Final + No Carveout + No Disclosures + PDF = Marketing Vendor Production

Draft + Yes Carveout + No Disclosures + Word = 2-SBCs (PDF and Word) - Draft
Draft + Yes Carveout + No Disclosures + PDF = 2-SBCs (PDF and Word) - Draft
Final + Yes Carveout + No Disclosures + Word = 2-SBCs (Word - Carve Out) - Final
Final + Yes Carveout + No Disclosures + PDF = Marketing Vendor Production

Draft + Yes Carveout + Yes Disclosures + Word = n/a
Draft + Yes Carveout + Yes Disclosures + PDF = 2-SBCs (PDF and Word) – Draft
Final + Yes Carveout + Yes Disclosures + Word = n/a
Final + Yes Carveout + Yes Disclosures + PDF = 1-SBCs (PDF) - Final

Draft + No Carveout + Yes Disclosures + Word = n/a
Draft + No Carveout + Yes Disclosures + PDF = 2-SBCs (PDF and Word) - Draft
Final + No Carveout + Yes Disclosures + Word = n/a
Final + No Carveout + Yes Disclosures + PDF = 1-SBCs (PDF) - Final

All other functionality originally put into place re: the Carve Out button has been removed such as graying out Disclosures button and not allowing PDF as an option when Carve Out = Yes. 

Please test at your convenience.
Thank you!

**Comment by willdwyer** [comment]
Hi Michelle,

A couple things...

First off, we've encountered a defect while testing.  See below screenshot and attached PDF.  The calculations for the diabetes calculator on this plan are incorrect.  This is for plan 3565160_Lisa Test_HMO_Blue_NE_Options_Ded_III_v.5_1234TEST.

![screen shot 2016-04-08 at 11 51 14 am](https://cloud.githubusercontent.com/assets/13453008/14389532/0b85c160-fd81-11e5-8756-483a5460b761.png)
[3565160_Lisa Test_HMO_Blue_NE_Options_Ded_III_v.5_1234TEST.pdf](https://github.com/willdwyer/bcbsmaissuestracker/files/210445/3565160_Lisa.Test_HMO_Blue_NE_Options_Ded_III_v.5_1234TEST.pdf)

**Comment by michparrish** [comment]
Hi Will,
Please open a new defect for that. That is not related to the code push. Thanks!

**Comment by willdwyer** [comment]
Michelle,

Also, after conferring internally, we would like to move this Rx Carve Out Plan radio button on the configurator so that it is below the Output option (Word or PDF).  Since this option is SOLELY used to determine whether or not a Word document should be mapped as 1-SBCs-Final (Word) or 2-SBCs Word – Carve Out) - Final, it should be contingent on whether or not a BE selects a Word or PDF option.  
Also, if a user does NOT select a Word Output option, the Rx Carve Out Plan radio button should be grayed out.

Please make this adjustment in PCLIENT5.

![screen shot 2016-04-08 at 12 20 32 pm](https://cloud.githubusercontent.com/assets/13453008/14390367/9672c716-fd84-11e5-8230-23bad552aad4.png)

**Comment by michparrish** [comment]
Will,
Those are additional requirements and will be submitted to Dev as such. They will respond with LOE and assessment.
As is, have you confirmed that the button is working as expected, where regardless of if Carve Out is selected or not, all files are routing to the correct folder?

**Comment by willdwyer** [comment]
So far, it seems the Carve Out functionality is working in PCLIENT5, but we still have a lot more testing to do.

And, we consulted again internally, and we would like to add these requirements to it.  The Rx Carve Out Plan radio button on the generator should be placed below the Output button, and should be grayed out unless the Word option is selected for Output.

Please let us know the additional LOE and assessment.

**Comment by michparrish** [comment]
Received notice today from the Development team that the next enhancement review cycle will be the first week of May. They will assess the priority of this item among all client enhancements.

My recommendation would be to move the code and configuration as is, if you've been able to test out the routing is correct, and we can enhance the location of the button and the 'graying out' at a later time, so you have this functionality in place sooner than later.
Please let us know.
Thank you.

**Comment by willdwyer** [comment]
Michelle,

So, you are saying that they wouldn't be able to change the coding for the Rx Carve Out Plan mapping button until first week of May?  Please provide a finite date when this would be completed by.

**Comment by michparrish** [comment]
Will - I'm saying the Dev team will not be reviewing this item (their prioritization meeting) until the first week of May. Only then can I provide more of an idea on when this item will be prioritized with all of the other items on Dev's plate and a date for you.

**Comment by willdwyer** [comment]
Michelle, we will discuss this on Friday, as I just discussed internally with Kevin.

Why aren't we seeing wire frames/UI infomation before enhancements go out?  Kevin brought this up at the last conference call we had.

**Comment by michparrish** [comment]
Hi Will,
I'm not sure what you mean. We discussed this last week and all agreed we would present samples prior to moving code, when a UI change is requested/needed. Since then, there has not yet been a need for a wireframe. Were you expecting a wireframe of something this week?

**Comment by willdwyer** [comment]
Hi Michelle,

Has HR started to review this yet? (it is the first week of May now)

Also, we discussed wireframes on the last weekly defect call (on Friday).  Tim/Mark indicated they would be provided to us in the coming weeks.

**Comment by michparrish** [comment]
Hi Will,

We reached out to the Dev team this morning to ask when they will be holding their meeting to evaluate all items in the queue. We'll keep you posted.
Thank you,
Michelle

**Comment by TimHeasley** [comment]
The dev team is working through a few critical top priorities and have not had the capacity to investigate the LOE to support the new requirement.  We hope to have a response possibly end of today, if not EOD early next week.

**Comment by TimHeasley** [comment]
Will,
Attached are the wire frames for the carve out.

Requirements:

> 1) Placement of the 'Rx Carve Out Yes/No' field to be below the Output options (Word, PDF) 
> 2) Rx Carve Out field to be grayed out unless the user has selected both *Mark Plan Document as: 'Final' and Output : 'Word'. When this combo is selected, the Rx Carve Out field should be enabled. 
> Thanks,
> Tim

[Prescription Carve-out requirements - Wireframe.docx](https://github.com/willdwyer/bcbsmaissuestracker/files/259836/Prescription.Carve-out.requirements.-.Wireframe.docx)

**Comment by willdwyer** [comment]
Hi Tim,

We reviewed these wire frames internally, and this works for us.  Please proceed with implementing the enhancement.

**Comment by TimHeasley** [comment]
Thank you for reviewing and approval.  We will move forward with the request.

**Comment by TimHeasley** [comment]
We have advised the development to proceed and let us know when the build takes place. After the build we will test in PClient5 and let you know when it is ready for retest in PClient5.

**Comment by TimHeasley** [comment]
Will,
We have moved the enhancement to PClient5 and it is ready for retest.
Let us know if you have any questions.
Thanks,
TIm

**Comment by willdwyer** [comment]
Encountered this during testing generation of a final carve out word document.

![screen shot 2016-05-20 at 9 48 29 am](https://cloud.githubusercontent.com/assets/13453008/15429672/0c694b6e-1e70-11e6-9cba-c28bd1a89aee.png)

**Comment by TimHeasley** [comment]
Hi Will,
Can you let me know the steps you took when you received the error?
I created a new plan using 3530621 from your example above as the base and I was able to generate a document (3579452).
Thanks,
Tim

**Comment by willdwyer** [comment]
1. Go to Content Library
2. Marketing and Sales/Custom/TJX folder
3. Selected checkbox next to plan design, clicked Edit, Go to Manage Plan
4. In configurator, made necessary steps to change to Carve Out (Plan Details, Carve Out - Yes, Misc Fields, Carve Out - Yes),
5. In configurator, changed Minimum Value Standard to Does Not Meet
6. Changed Minimum Creditable Coverage to Does Not Meet
7. Saved/Validated
8. Clicked Plan Documents button
9. Entered Final, Word, Carve Out - Yes radio mapping button
10. Generated

**Comment by willdwyer** [comment]
I'm also getting this error when trying to create a Custom from a standard in the configurator.

![screen shot 2016-05-20 at 10 10 18 am](https://cloud.githubusercontent.com/assets/13453008/15430455/6e9bcc64-1e73-11e6-8d14-b83c288982d9.png)

**Comment by willdwyer** [comment]
Just got it again.  See screenshot.

Same steps, except see the plan I used in the screenshot.

![screen shot 2016-05-20 at 10 32 05 am](https://cloud.githubusercontent.com/assets/13453008/15431054/2514e244-1e76-11e6-824b-3580abc753a8.png)

**Comment by TimHeasley** [comment]
Thanks for sending all of the details.  We are working to get the permissions fixed. We'll let you know as soon as it is completed.

**Comment by TimHeasley** [comment]
Will,
We are researching the issue, were you using IE10 at the time of the issue?
Thanks,
Tim

**Comment by willdwyer** [comment]
Yes, we were using Ie10.

**Comment by willdwyer** [comment]
Assigning to Michelle, as I know Tim is on vacation.

**Comment by michparrish** [comment]
Hi Will,

Permissions issues have been fixed in pclient5 and I retested the new code, logged in as You, and it tested successfully.
Please continue with your testing and let me know how it goes.
Thanks!
michelle

**Comment by willdwyer** [comment]
Not fully implemented.  

After testing an HCCS plan (specifically 800034HR_HMO_Blue_NE_Ded_1000_Coins_80-0263HCCSSG1-1-16), the Carve Out logic did not apply correctly. Per screen shot, the Highest Cost Share columns still has cost share information.  Please make this fix to all plans.

![screen shot 2016-05-25 at 10 32 11 am](https://cloud.githubusercontent.com/assets/13453008/15547523/278f1008-2272-11e6-9764-82a5c9747f1c.png)

**Comment by michparrish** [comment]
Hi Will,
I see the plan you used. Did you use the Carve Out Yes/No fields within the configurator on this? Those are what impact the data itself.
Please send me the steps you took to update the copy of 800034HR so I can replicate.
Thank you!

**Comment by willdwyer** [comment]
Michelle,

The steps were as follows...
1. Go to Manage Plan
2. Search for:
   Effective Date - 2016
   Product Group Type - HMO Blue NE
   Plan Type - Standard
3. Hover over plan 800034HR_HMO_Blue_NE_Ded_1000_Coins_80-0263HCCSSG1-1-16
4. Click Create a New Custom
5. Plan Details tab - Carve Out, yes
6. Misc Fields tab - Carve Out, yes
7. Proceed to Save/Validate
8. Carve Out radio mapping button, Yes
9. Final, Word
10. Generate

**Comment by michparrish** [comment]
Thanks! I was able to recreate. I've passed this along to update the workbook(s).

**Comment by michparrish** [comment]
Ok, all set. All workbooks have been updated.
Please begin retesting again and let me know how it goes.
Thanks!

**Comment by willdwyer** [comment]
Hi Michelle,

I am still encountering errors.  When entering Yes for Carve Out in the Misc Fields tab, it does not seem to be saving.  See screenshot below for the Advantage Blue plan I tested that still has cost share amounts for Rx drug information.

Please note I tested this three times over to confirm.

![screen shot 2016-05-26 at 4 59 23 pm](https://cloud.githubusercontent.com/assets/13453008/15590017/76120a9a-2363-11e6-8419-5643236f1024.png)

**Comment by michparrish** [comment]
Alright - the EPO workbook has been updated and retested successfully. Please let me know if you run into any other issues. Thanks!

**Comment by willdwyer** [comment]
Still more issues...

Tried creating a carve out from a PPO Blue Options plan (800168HR_PPO_Blue_Options_v.5_80-0222MMASCBF1-1-16) and the Carve Out button is missing from the Plan Details tab.  See screenshot.

![screen shot 2016-05-27 at 12 25 03 pm](https://cloud.githubusercontent.com/assets/13453008/15614330/3113a11c-2406-11e6-9368-72c5d94c6e04.png)

**Comment by michparrish** [comment]
that is odd. I see it when I pull up a different plan, but not this one. I'm going to send to our QA dept. There should be no reason why an attribute is missing from this page. Additionally, I did make sure that the attribute for this plan had a value, in case a missing value was the reason it wouldn't show. It does show that the value for this plan is 'No' for the Carve Out attribute.

**Comment by willdwyer** [comment]
@mattrimer

**Comment by michparrish** [comment]
this is all set - workbook has been updated. It was missing the 'Carve Out' attribute row in one of the tabs. I've confirmed this row is accounted for in all of the other workbooks.
Please continue retesting. Thanks!

**Comment by mattrimer** [comment]
After retesting on PClient5 I found that this issue still exists. The Carve Out button is missing from the Plan Details tab.

See screenshot:

![screen shot 2016-05-31 at 1 57 21 pm](https://cloud.githubusercontent.com/assets/18705500/15685022/e1c5f166-2737-11e6-898c-1642357ad8c3.png)

**Comment by michparrish** [comment]
Hi Matt,
Thanks for retesting - I did a search on that exact plan and called it into the configurator and I see the Rx Carve Out row. Would you mind trying to either clear your cache or refresh? Please be sure you are also using IE10. Please let me know if this works. Thank you!
![rx carve out attribute](https://cloud.githubusercontent.com/assets/13471811/15685295/0ec12838-2739-11e6-9eca-bcf2c8a962e9.JPG)

**Comment by mattrimer** [comment]
The previous issue regarding the carveout option not showing up in the plan detail screen has been resolved.

However, I found that an issue from last week still exists. The Misc fields tab carveout option still resets to "no" after the word doc has been generated on certain plans. From what I've sampled, I found that Blue Choice local $10 (800202HR) and Blue Choice local Plan 2 (800305HR) have this problem, but I suspect there are more within the Blue choice group that also have this issue. For that reason, I think it would be wise for your team to make sure all plans in this group have been fixed.

![screen shot 2016-06-01 at 9 57 28 am](https://cloud.githubusercontent.com/assets/18705500/15713622/2bdf114e-27e5-11e6-9df7-d31eb0fe46e8.png)

![screen shot 2016-06-01 at 9 56 30 am](https://cloud.githubusercontent.com/assets/18705500/15713641/3cd2f0d8-27e5-11e6-890d-6a5eb0408fe9.png)

**Comment by TimHeasley** [comment]
Matt,
Thank you for testing. I am also seeing that the Rx Carve Out is reverting back to "no" on the Misc Tab.  I will research the issue and let you know when you can retest.
Thanks,
Tim

**Comment by TimHeasley** [comment]
Matt, 
We have updated the workbooks, tested & confirmed that the values do not revert for any plans.  Ready for retest in PClient5.
Thanks,
Tim

**Comment by mattrimer** [comment]
Thanks Tim,

I've retested these plans and found no errors. This item can be pushed to the next step.

**Comment by TimHeasley** [comment]
Update is being pushed to Production environment the evening of 6/3 we will verify updates and let BCBSMA team know when they can test on Monday.

**Comment by TimHeasley** [comment]
The enhancement has been moved to Production and is ready for review by the BCBSMA team.
Thanks, Tim

**Comment by mattrimer** [comment]
Looks good, this request is closed.
